lib/types/wadl.js

```diff
@@ -57,24 +57,35 @@ function convertToSwagger(wadl, callback) {
 
     //HACK: we just strip namespace. Yes I know, it's ugly.
     //But handling XML namespaces is even uglier.
-    var match = wadlType.match('^[^:]\+:(.+)$');
-    assert(match);
+    var match = wadlType.match(/^(?:[^:]+:)?(.+)$/);
+    assert(match, wadlType);
     var type = match[1];
-    switch (type) {
+    switch (type.toLowerCase()) {
       case 'boolean':
       case 'string':
       case 'integer':
         return {type: type};
       case 'double':
-        return {type: "number"};
+      case 'decimal':
+        return {type: 'number'};
       case 'int':
-        return {type: "integer", minimum: -2147483648, maximum: 2147483647};
+        return {type: 'integer', minimum: -2147483648, maximum: 2147483647};
       case 'long':
-        return {type: "integer", minimum: -9223372036854775808, maximum: 9223372036854775807};
+        return {type: 'integer', minimum: -9223372036854775808, maximum: 9223372036854775807};
```

Comment:
Sounds good. My point was that if we actually hit this case, the JSON produced won't have the right number since it was produced by this JS.

FWIW, it looks like JS just cuts the number at MAX_SAFE, at least in Firefox/SpiderMoney.

```js
console.log(9223372036854775807)
> 9223372036854776000
```